lib/pharos/kube/resource.rb

```diff
@@ -0,0 +1,86 @@
+# frozen_string_literal: true
+require 'kubeclient'
+require 'pathname'
+
+module Pharos
+  module Kube
+    class Resource
+      def initialize(session, resource, vars = {})
+        @resource =
+          case resource
+          when Kubeclient::Resource
+            resource
+          when Hash
+            Kubeclient::Resource.new(resource)
+          else
+            Kubeclient::Resource.new(Pharos::YamlFile.new(resource).load(vars))
```

Comment:
This goes back to the excessive type confusion thing again... IMO it's not a good design to be way too flexible in terms of the variety of different types accepted.

This is demonstrated by the `vars` getting totally ignored unelss you pass something that's a `String` :-1: 

These might make sense as separate class methods, like `def self.resource_from_file(file, vars = {})`